src/Raven.Client/Documents/Commands/Batches/ICommandData.cs

```diff
@@ -11,8 +10,17 @@ public interface ICommandData
 
         long? Etag { get; }
 
-        HttpMethod Method { get; }
+        CommandType Type { get; }
 
         DynamicJsonValue ToJson(DocumentConventions conventions, JsonOperationContext context);
     }
+
+    public enum CommandType
+    {
+        None,
+        PUT,
+        PATCH,
+        DELETE,
+        AttachmentPUT
```

Comment:
What about removing an attachment?

What about rename? Do we have kind of support for that?
What about moving an attachment from one document to another? 
What about copying an attachment (server side only op)?

Created an issue for that http://issues.hibernatingrhinos.com/issue/RavenDB-6976